### PR TITLE
feat(OnyxListbox): implement select

### PR DESCRIPTION
**Pull request by JoCa96**
Relates to #565, #706

- update OnyxListbox with select/combox features
- listbox will be renamed to `OnyxSelect` in a follow up PR
- fix `OnyxSelectInput` styles for hover, focus etc.

## Checklist

- [x] The added / edited code has been documented with [JSDoc](https://jsdoc.app/about-getting-started)
- [x] If a new component is added, at least one [Playwright screenshot test](https://github.com/SchwarzIT/onyx/actions/workflows/playwright-screenshots.yml) is added
- [x] A changeset is added with `npx changeset add` if your changes should be released as npm package (because they affect the library usage)

### CODE REVIEW

File: apps/playground/src/components/VersionSelect.vue [hunk 1]
```diff
@@ -85,7 +86,13 @@ onBeforeUnmount(() => {
     />
 
     <!-- TODO: remove when OnyxSelect is fully implemented and add loading/empty state -->
-    <OnyxListbox v-if="expanded" v-model="modelValue" label="Select version" :options="options" />
```
Comment (JoCa96) [inline apps/playground/src/components/VersionSelect.vue:88]: Remove OnyxSelectInput

File: packages/sit-onyx/src/components/OnyxListbox/OnyxListbox.stories.ts [hunk 2]
```diff
@@ -72,23 +74,10 @@ const meta: Meta<typeof OnyxListbox> = {
...
-  { value: "dog", label: "Dog", group: "Land" },
-  { value: "tiger", label: "Tiger", group: "Land" },
-  { value: "reindeer", label: "Reindeer", group: "Land" },
-  { value: "racoon", label: "Racoon", group: "Land" },
-  { value: "dolphin", label: "Dolphin", group: "Water" },
-  { value: "flounder", label: "Flounder", group: "Water" },
-  { value: "eel", label: "Eel", group: "Water" },
-  { value: "falcon", label: "Falcon", group: "Air" },
-  { value: "owl", label: "Owl", group: "Air" },
-];
```
Comment (JoCa96) [inline packages/sit-onyx/src/components/OnyxListbox/OnyxListbox.stories.ts:86]: readd

File: packages/sit-onyx/src/components/OnyxListbox/OnyxMiniSearch.stories.ts [hunk 3]
```diff
@@ -9,7 +9,7 @@ const meta: Meta<typeof OnyxMiniSearch> = {
   title: "support/MiniSearch",
   ...defineStorybookActionsAndVModels({
     component: OnyxMiniSearch,
-    events: ["update:modelValue"],
```
Comment (JoCa96) [inline packages/sit-onyx/src/components/OnyxListbox/OnyxMiniSearch.stories.ts:12]: readd "update:modelValue"

File: packages/sit-onyx/src/components/OnyxListbox/OnyxListbox.vue [hunk 4]
```diff
@@ -181,93 +241,106 @@ watchEffect(() => {
...
-          v-bind="headlessGroup({ label: group })"
-        >
-          <li
-            v-if="group != ''"
-            role="presentation"
-            class="onyx-listbox__group-name onyx-text--small"
-          >
-            {{ group }}
+  <div class="onyx-combobox-wrapper">
+    <OnyxSelectInput
```
Comment (JoCa96) [inline packages/sit-onyx/src/components/OnyxListbox/OnyxListbox.vue:245]: Add remaining OnyxSelectInput config Props

File: packages/sit-onyx/src/components/OnyxListbox/OnyxListbox.vue [hunk 5]
```diff
@@ -52,85 +58,133 @@ const slots = defineSlots<{
...
 
-const searchInput = computed({
-  get: () => (props.withSearch && props.searchTerm) || "",
-  set: (newValue) => emit("update:searchTerm", newValue),
+const arrayValue = computed(() => {
+  if (!props.modelValue) {
+    return [];
+  }
+  return props.multiple ? props.modelValue : [props.modelValue];
 });
```
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxListbox/OnyxListbox.vue:80]:
```suggestion
const arrayValue = computed(() => {
  if (!props.modelValue) return [];
  return props.multiple && Array.isArray(props.modelValue) ? props.modelValue : [props.modelValue];
});
```

The potential error that you showed me earlier when playing around with the Storybook controls could maybe be fixed with this minor change

File: packages/sit-onyx/src/components/OnyxListbox/OnyxListbox.vue [hunk 6]
```diff
@@ -52,85 +58,133 @@ const slots = defineSlots<{
...
-  multiple: computed(() => !!props.multiple),
-  activeOption,
-  onSelect: (selectedOption) => {
-    if (selectedOption === CHECK_ALL_ID) {
-      checkAll.value?.handleChange(!checkAll.value.state.value.modelValue);
-      return;
-    }
+const onToggle = () => {
+  isExpanded.value = !isExpanded.value;
+  if (!isExpanded.value) {
```
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxListbox/OnyxListbox.vue:109]: We could also move the watchEffect above into here
image

File: packages/sit-onyx/src/components/OnyxListbox/OnyxListbox.vue [hunk 7]
```diff
@@ -52,85 +58,133 @@ const slots = defineSlots<{
...
+
+const onSelect = (selectedOption: string) => {
+  if (selectedOption === CHECK_ALL_ID) {
+    checkAll.value?.handleChange(!checkAll.value.state.value.modelValue);
+    return;
+  }
+  const newValue = props.options.find(({ value }) => value === selectedOption);
+  if (!newValue) {
+    return;
+  }
```
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxListbox/OnyxListbox.vue:149]: Like this de-selecting a value does not work, right? Shouldn't we support this?

File: packages/sit-onyx/src/components/OnyxListbox/OnyxListbox.vue [hunk 8]
```diff
@@ -181,93 +241,106 @@ watchEffect(() => {
...
+        <OnyxMiniSearch
+          v-if="props.withSearch"
+          ref="miniSearch"
+          v-bind="input"
+          :label="t('listbox.searchInputLabel')"
+          class="onyx-listbox__search"
+          @clear="emit('update:searchTerm', '')"
+        />
+
+        <ul v-if="isEmptyMessage" role="group" aria-label="" class="onyx-listbox__group">
```
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxListbox/OnyxListbox.vue:269]: Is the empty aria label intended?

File: packages/sit-onyx/src/components/OnyxListbox/OnyxListbox.vue
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxListbox/OnyxListbox.vue:1]: Do you already want to rename the listbox to select in this PR?

**Review by larsrickert — CHANGES_REQUESTED** [review]

File: packages/sit-onyx/src/components/OnyxListbox/OnyxListbox.stories.ts [hunk 9]
```diff
@@ -162,16 +146,6 @@ export const Multiselect = {
   },
 } satisfies Story;
 
-/**
```
[the remark this answers is not in the document]
Comment (JoCa96) [reply packages/sit-onyx/src/components/OnyxListbox/OnyxListbox.stories.ts:165]: nope, was readded

File: packages/sit-onyx/src/components/OnyxListbox/OnyxListbox.vue (hunk 7 above)
Comment (JoCa96) [reply packages/sit-onyx/src/components/OnyxListbox/OnyxListbox.vue:149]: It works, `onSelect` is triggered whenever a option is clicked, even when it's already selected.

Comment (JoCa96) [reply packages/sit-onyx/src/components/OnyxListbox/OnyxListbox.vue:149]: For a single select, the deselection is not allowed.

**Review by MajaZarkova — APPROVED** [review]

**Review by larsrickert — APPROVED** [review]

**Review by MajaZarkova — APPROVED** [review]

**Pull request merged**